### PR TITLE
New C# subscriber

### PR DESCRIPTION
**Pull request by Arashrohani**
# Description

I have created a new subscriber for c# in pubsub sample. I think this sample would be helpful for people who wants to start with asp.net core. In topic C have utilized dictionary to show different way of receiving message.
I have tested locally and it works.

I have updated the README file to the best of my knowledge but couldn't update the pictures (I hope you guys can help with that).

besides, I have added the yaml file for kubernetes deployment but couldn't test it because I don't have access to AKS.

Please let me know if I need to do anything else for this PR to be merged.

### CODE REVIEW

**Comment by wcs1only** [comment]
So in general we have sdk specific examples of dapr building blocks hosted with the SDKs in general. For the dotnet-sdk, you can find the examples here: 

https://github.com/dapr/dotnet-sdk/tree/master/examples

For now I think adding a dotnet-sdk subscriber to this example complicates it a bit more than we'd like. We're currently discussion a larger effort that may reimplement the quickstarts using each one of the SDKs so that a dotnet developer can run the same examples native to dotnet as a python developer does, as a node developer does, and so on.

**Comment by Arashrohani** [comment]
Thank you for your response, The beauty of this sample was using multiple technologies and I was excited to add C# version because I saw some questions about pub-sub in discord channel but if you guys are not interested I understand.

**Comment by Arashrohani** [comment]
Please let me know if you want me to close this PR?

**Comment by paulyuk** [comment]
Hi @Arashrohani, thank you so much for this contribution.  I like it.  LGTM.  @msfussell could you please consider approving and merging?  

Note these quickstarts will be moved to a repo called `tutorials`.  I think this is a nice addition.

You might also be interested in the new approach we have to quickstarts in the works.   This will mean that every feature and API will be done well in all major languages, including C#.  We're even showing the .NET 6 minimal API.  Could you please check this out?  https://github.com/dapr/quickstarts/tree/feature/new_quickstarts

**Review by paulyuk — APPROVED** [review]
LGTM.  Note I see this moving along with this repo to `tutorials`.

**Review by paulyuk — CHANGES_REQUESTED** [review]
LGTM, just need you to pull and merge changes, and then push again.  Let's also make sure tests run and pass. thank you.

**Comment by Arashrohani** [comment]
Hi @paulyuk, Thank you.
I updated this branch with the base branch, what else should I do?
It says that I need to add sign off by author, Do you want me to do that?
Is Tests are going to run as part of pull request process or should I run them manually?
Thanks.

**Comment by Arashrohani** [comment]
@paulyuk Hey Paul, I followed the instruction for DCO and updated the branch again. I'm waiting for your instruction for next step.

**Comment by paulyuk** [comment]
@Arashrohani Got it thank you much for doing the process.  Ok, I unblocked tests for you.  Let's give the tests workflow about 30 mins to see results!  If that looks good I think you'll see a looks good.

**Comment by Arashrohani** [comment]
@paulyuk I updated the branch again, would you please run the tests again? I don't know what was the issue first time.

**Comment by Arashrohani** [comment]
@paulyuk It seems that the order of some outputs has changed for calculator and causing the test to fail as far as I could understand from the logs.

**Comment by Arashrohani** [comment]
Hi @paulyuk I was wondering what is gonna happen to this PR?

**Comment by amulyavarote** [comment]
@Arashrohani Can you please pull the latest from master and rebase this?

**Comment by Arashrohani** [comment]
@amulyavarote It's Done.

**Comment by davidfowl** [comment]
Can we make this .NET 6 minimal API based instead?

**Comment by Arashrohani** [comment]
@davidfowl I was thinking about that also but wasn't sure when Should I do it? Now or after this branch was merged?

**Comment by davidfowl** [comment]
I would do it now.

**Comment by Arashrohani** [comment]
@davidfowl Yeah It makes sense, I will upgrade to .NET 6 minimal API and push it up.

**Comment by Arashrohani** [comment]
@davidfowl @amulyavarote It's been upgraded to ASP.NET Core 6 Minimal API.

**Comment by amulyavarote** [comment]
> @davidfowl @amulyavarote It's been upgraded to ASP.NET Core 6 Minimal API.

Can you please fix the build?

**Comment by Arashrohani** [comment]
@amulyavarote Sorry about that, I just noticed that README file it's being used for testing also, I added csharp-subscriber to the makefile and update the README based on the outputs from the tests, hopefully these changes will fix it but How can I re-run the tests?

**Comment by amulyavarote** [comment]
> @amulyavarote Sorry about that, I just noticed that README file it's being used for testing also, I added csharp-subscriber to the makefile and update the README based on the outputs from the tests, hopefully these changes will fix it but How can I re-run the tests?

Yes! You can!

**Comment by Arashrohani** [comment]
@amulyavarote Would you please let me know how? I thought that I can't do it because of this: "First-time contributors need a maintainer to approve running workflows".

**Comment by Arashrohani** [comment]
@amulyavarote Would you please approve running the workflows again?

**Comment by amulyavarote** [comment]
@Arashrohani I reran the workflows. Tutorials workflow is still failing. Can you please check that?
https://github.com/dapr/quickstarts/runs/6373558996?check_suite_focus=true

**Comment by Arashrohani** [comment]
@amulyavarote How can I push my image to ghcr.io? I think the issue with K8s test is that my image has not been pushed to the registry (I don't think that I have access to do that myself), would you please help me on that?

File: tutorials/pub-sub/README.md
Comment (davidfowl) [inline tutorials/pub-sub/README.md:584]: Do we need the swagger?

Comment (Arashrohani) [reply tutorials/pub-sub/README.md:584]: @davidfowl I don't think so, it came with default template and I didn't remove it, Do you want me to remove it?

Comment (davidfowl) [reply tutorials/pub-sub/README.md:584]: Yep. Lets make this minimal 😄

**Comment by Arashrohani** [comment]
@davidfowl @amulyavarote Removed the Swagger and added the missing steps in README file and tested locally with mm tool and everything passed except the K8s deployment because of the image, I tried to test that locally also but React-form was throwing exception, would you please re-run the workflows to test it?

File: tutorials/pub-sub/csharp-subscriber/Program.cs
Comment (davidfowl) [inline tutorials/pub-sub/csharp-subscriber/Program.cs:11]: This is also built in so you can remove it.

File: tutorials/pub-sub/README.md
Comment (davidfowl) [inline tutorials/pub-sub/README.md:643]: @rynowak Should we have a method on IEndpointConventionBuilder that adds this metadata as well?

File: tutorials/pub-sub/csharp-subscriber/Program.cs
Comment (Arashrohani) [reply tutorials/pub-sub/csharp-subscriber/Program.cs:11]: You mean the whole if section can be removed?

Comment (davidfowl) [reply tutorials/pub-sub/csharp-subscriber/Program.cs:11]: Yep

**Review by davidfowl — APPROVED** [review]
LGTM

**Comment by Arashrohani** [comment]
@davidfowl Thanks, I don't know how to push csharp-subscriber image to the registry and I think that's the reason for failing tests. I can see the code for push in docker.mk file but when I run that code it just builds the images (I think it's related to some sort of access or something). @amulyavarote @paulyuk How can I push my image to the registry?

**Comment by amulyavarote** [comment]
@Arashrohani Hi, Can you please rebase the changes with master and push the changes to fix the build?

**Comment by Arashrohani** [comment]
@amulyavarote Hi, Sure, I will do that but as I mentioned before I don't think that fixes the build, I just checked and this is still missing: ghcr.io/dapr/samples/pubsub-csharp-subscriber:latest
this image has been used in csharp-subscriber.yaml and imagePullPolicy is always and I think because that image doesn't exist the build can not deploy the csharp-subscriber and fails.

**Comment by amulyavarote** [comment]
> @amulyavarote Hi, Sure, I will do that but as I mentioned before I don't think that fixes the build, I just checked and this is still missing: ghcr.io/dapr/samples/pubsub-csharp-subscriber:latest this image has been used in csharp-subscriber.yaml and imagePullPolicy is always and I think because that image doesn't exist the build can not deploy the csharp-subscriber and fails.

Got it! Let me check!

**Comment by Arashrohani** [comment]
@amulyavarote I updated the code and pushed it up, I noticed that one step "publish docker manifest" is being skipped and I guess that's why the image is missing, would you please push it up and rerun the tests?

**Comment by paulyuk** [comment]
> @amulyavarote I updated the code and pushed it up, I noticed that one step "publish docker manifest" is being skipped and I guess that's why the image is missing, would you please push it up and rerun the tests?

Hey @Arashrohani thank you much for this contribution.  Really nice.  You nailed it - we need your change to the makefile to get into the branch for the upload to happen.  This is a bit of a chicken and an egg situation.  Technically pushing the makefile change first as a separate PR is the ideal way to go.  But given low risk nature, I'm happy to just merge this and then we "test in prod".  You're cool to work with us to get tests passing after?

**Comment by Arashrohani** [comment]
@paulyuk Hi Paul, Thanks, Yes, Of course I will work with you guys to get tests passing.

**Pull request merged**
